src/Lecture3.hs

```diff
@@ -79,7 +86,9 @@ Tuesday
   would work for **any** enumeration type in Haskell (e.g. 'Bool',
   'Ordering') and not just 'Weekday'?
 -}
-next = error "TODO"
+next :: Weekday -> Weekday
+next Sunday = Monday
```

Comment:
That looks great! 👏🏻 
If you want to solve the bonus challenge, you can compare the value of `d` with `maxBound` and return `minBound` in this case. And after you should be able to make the type more polymorphic and allow this function to work with any enumeration 🙂